New post: CSS `accent-color` (#5952)
GoogleChrome/web.dev: src/site/content/en/blog/accent-color/index.md

---
layout: post
title: CSS `accent-color`
subhead: Bring your brand color to built-in HTML form inputs with one line of code.
authors:

>   - adamargyle

Reviewer: would it be cool if i'm an author too...?

Author: of course!

Reviewer: https://github.com/GoogleChrome/web.dev/pull/5953

Reviewer 2: I approved the PR adding you, you'll need to merge the main branch into this one I think, otherwise, the build will complain.

Author: ![Screen Shot 2021-08-10 at 9 07 13 AM](https://user-images.githubusercontent.com/1134620/128896936-3272d45a-d4a8-4f8d-9994-fe62ec92aa3f.png)

Author: github is being wierd today and the commits say they pushed but arent reflected in this PR yet

description: Bring your brand color to built-in HTML form inputs with one line of code.
date: 2021-08-09
thumbnail: image/vS06HQ1YTsbMKSFTIPl2iogUQP73/WOcuCLCwMr0M2lF17bmm.png
hero: image/vS06HQ1YTsbMKSFTIPl2iogUQP73/huEpiCoJQ6dAo8rHGsZT.png
tags:
  - blog
  - css
---

> Today's HTML form elements are [difficult to
> customize](https://codepen.io/GeoffreyCrofte/pen/BiHzp). It feels as if it's
> either: no custom styles, very few custom styles, or reset input styles and
> build it up from scratch. Building it up from scratch ends up being much more
> work than anticipated; forgotten styles for element states
> ([indeterminate](https://developer.mozilla.org/en-US/docs/Web/CSS/:indeterminate),
> I'm looking at you), losing built in accessibility features, etc. It may be more
> work than you're looking to take on.

Reviewer 2: ```suggestion
Today's HTML form elements are [difficult to
customize](https://codepen.io/GeoffreyCrofte/pen/BiHzp). It feels as if it's
a choice between few or no custom styles, or resetting input styles and
build it up from scratch. Building it up from scratch ends up being much more
work than anticipated. It can also lead to forgotten styles for element states
([indeterminate](https://developer.mozilla.org/en-US/docs/Web/CSS/:indeterminate),
I'm looking at you), and the loss of built-in accessibility features. 
To fully recreate what the browser provides may be more
work than you're looking to take on.
```

```css
accent-color: hotpink;
```

CSS `accent-color` from the [CSS UI
specification](https://www.w3.org/TR/css-ui-4/#widget-accent) is here to tint
elements with one line of CSS, saving you from customization efforts by
providing a way to bring your brand into elements.

<figure class="w-figure ">
  {% Img 
    src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/CfSS3F1XUsfCHIB86xeE.png", 
    class="w-screenshot",
    alt="A light theme screenshot of an accent-color demo where 
    checkbox, radio buttons, a range slider and progress element 
    are all tinted hotpink.", 
    width="800", height="548" 
  %}
  <figcaption class="w-figure">
    <a href="https://codepen.io/web-dot-dev/pen/PomBZdy">Demo</a>
  </figcaption>
</figure>

The `accent-color` property also works with

> [`color-scheme`](https://web.dev/color-scheme/), allowing authors to tint both

Reviewer 2: ```suggestion
[`color-scheme`](/color-scheme/), allowing authors to tint both
```

> the light and dark elements. Here the user has dark theme active, the page uses

Reviewer 2: ```suggestion
the light and dark elements. 
In the following example the user has a dark theme active, the page uses
```

`color-scheme: light dark`, and uses the same `accent-color: hotpink` for dark
themed hotpink tinted controls.

<figure class="w-figure">
  {% Img 
    src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/3gxeeZoSLY34tsMxkyt9.png", 
    alt="A dark theme screenshot of an accent-color demo where 
    checkbox, radio buttons, a range slider and progress element 
    are all tinted hotpink.", 
    width="800", height="548" 
  %}
  <figcaption class="w-figure">
    <a href="https://codepen.io/web-dot-dev/pen/PomBZdy">Demo</a>
  </figcaption>
</figure>

### Browser support

As of this writing, Chromium 93+ and Firefox 92+ support `accent-color`.

> ## Supported Elements

Reviewer 2: ```suggestion
## Supported elements
```

Currently, only four elements will tint via the `accent-color` property:
[checkbox](#checkbox), [radio](#radio), [range](#range) and
[progress](#progress). Each can be previewed here
[https://accent-color.glitch.me](https://accent-color.glitch.me) in light and
dark color schemes.

{% Aside "warning" %}
If the following demo elements are all the same color, 
then your browser doesn't support `accent-color` yet.
{% endAside %}

### Checkbox

{% Codepen {
  user: 'web-dot-dev',
  id: 'dyWjGqZ'
} %}

### Radio

{% Codepen {
  user: 'web-dot-dev',
  id: 'WNjKrgB'
} %}

### Range

{% Codepen {
  user: 'web-dot-dev',
  id: 'yLbqeRy'
} %}

### Progress

{% Codepen {
  user: 'web-dot-dev',
  id: 'rNmrxqL'
} %}

> ## Guaranteeing Contrast

Reviewer 2: ```suggestion
## Guaranteeing contrast
```

To prevent inaccessible elements from existing, browsers with `accent-color`
need to determine an [eligible contrast
color](https://webaim.org/articles/contrast/) to be used alongside the custom
accent. Below is a screenshot demonstrating how Chrome 94 (left) and Firefox 92
Nightly (right) differ in their algorithms:

{% Img 
  src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/DJhB56n10Eh8O29RsRdE.png", 
  class="w-screenshot",
  alt="A screenshot of Firefox and Chromium side by side, 
  rendering a full spectrum of checkboxes in various hues and darknesses.", 
  width="800", height="832" 
%}

The most important thing to take away from this, is to **trust the browser**.
Provide a brand color, and trust that it will make smart decisions for you.

{% Aside %}
The browser will not change your color in a dark theme.
{% endAside %}

> ## Extra: more tinting

Reviewer 2: ```suggestion
## Extra: More tinting
```
